lua/core/midi.lua

```diff
@@ -203,6 +236,7 @@ function Midi.to_data(msg)
 end
 
 --- convert data (midi bytes) to msg
+-- @tparam data
```

Comment:
and type here too?

Reply:
so something like this:

```
--- convert msg to data (midi bytes).
-- @tparam table msg : 
-- @return hex : midi status bytes - command and value


--- convert data (midi bytes) to msg.
-- @tparam table data :
-- @return table msg : midi message table, contents vary depending on message
```

Reply:
right.  looking at https://github.com/luarocks/lua-style-guide#documentation, it seems the convention might be `<var> <type>: description` though we should confirm w/ ldoc proper.

Reply:
ok, hmmm. ldoc has examples that are a little different.  for example:

https://stevedonovan.github.io/ldoc/examples/four.lua.html

i don't feel super strongly to be honest but, like you, think we should just settle on something and then try and be consistent.

i'd be curious what @artfwo, @tehn, @catfact  and others think...

Reply:
@pq seems like ldoc uses `@ <name> <description>` by default, see chapter "Tags" [here](https://stevedonovan.github.io/ldoc/manual/doc.md.html).
Don't really see if that can include type info.

Apparently they also have an "alternative" `-- type: <name> <description>` way, see "Alternative way of specifying tags".

P.S. Sorry, but I can't link to specific headings/chapters because no anchors are included in ldoc's own docs :man_facepalming: 
Hope that's not something that carries over to the docs generated with ldoc.

Reply:
aha.  sorry yeah.  it looks like i was confusing tools (and advice) and maybe @okyeron was spot on to begin with.  sorry for the noise...

@simonvanderveldt: i'm curious if otherwise you have any preferences on standards?

Reply:
No strong opinion, I've done very little in Lua so don't really have a clue :P Mainly work in Python where I'm mostly used to either Google's style or reStructuredText (some examples here https://stackoverflow.com/a/24385103).

Do you/we want to include type information in the docstrings? It could be I've missed it but it seems like the `@ <name> <description>` variant doesn't support type info?